React/CoreModules/RCTAlertController.m

```diff
@@ -29,6 +29,9 @@ - (UIWindow *)alertWindow
 
 - (void)show:(BOOL)animated completion:(void (^)(void))completion
 {
+  if (@available(iOS 13.0, *)) {
+    self.overrideUserInterfaceStyle = UIApplication.sharedApplication.delegate.window.overrideUserInterfaceStyle;
```

Comment:
Instead of `UIApplication.sharedApplication`, can yo use `initWithFrame:RCTSharedApplication()`?
The UIApplication is not always available (in extensions, for example).

What happens if `UIApplication.sharedApplication.delegate.window.overrideUserInterfaceStyle;` returns `nil`?

Reply:
@cipolleschi I have refactored the code to use `RCTSharedApplication()`

> What happens if ...overrideUserInterfaceStyle; returns nil?

I don't believe that should be happening; it'd rather be `UIUserInterfaceStyleUnspecified`. But to be on the safe side of things, I made the assignment default to `UIUserInterfaceStyleUnspecified` in case `RCTSharedApplication()` were to return `nil`.

Reply:
Unfortunately, if any part of the chain is `nil`, it can happen due to [opional chaining](https://docs.swift.org/swift-book/LanguageGuide/OptionalChaining.html) - this is in Swift, but it works in the same way for Objective-C.

If `RCTSharedApplication` return `null`, the final object returned will be `null`, for example. In Objective-C, you can call any method on a `null` instance and get a `null` in return! 😄

Reply:
> Unfortunately, if any part of the chain is nil, it can happen

yes, I'm aware of that 🙂 and it's why I made the assignment default to `UIUserInterfaceStyleUnspecified`.

maybe I don't understand extensions enough, I just thought this code wouldn't be called in an extension, but I can be wrong 😄 
there is also this part which I thought would handle the nil case more explicitly if there was a risk of having `nil`  `RCTSharedApplication()`. Is that part okay from that point of view? Thanks.

https://github.com/facebook/react-native/blob/6fcfe2e1b35e9bdf319bcdc1647c8c0d997b58c7/React/CoreModules/RCTAlertController.m#L23